packages/components/checkbox/src/CheckboxGroup.tsx

```diff
@@ -0,0 +1,80 @@
+import { useFormFieldState } from '@spark-ui/form-field'
+import { useCombinedState } from '@spark-ui/use-combined-state'
+import { ComponentPropsWithoutRef, forwardRef, useEffect, useMemo, useRef } from 'react'
+
+import { checkboxGroupStyles, CheckboxGroupStylesProps } from './CheckboxGroup.styles'
+import { CheckboxGroupContext, CheckboxGroupContextState } from './CheckboxGroupContext'
+
+export interface CheckboxGroupProps
+  extends Omit<ComponentPropsWithoutRef<'div'>, 'value' | 'defaultValue' | 'onChange'>,
+    CheckboxGroupStylesProps,
+    Pick<CheckboxGroupContextState, 'intent' | 'name' | 'value'> {
+  /**
+   * The initial value of the checkbox group
+   */
+  defaultValue?: string[]
+  /**
+   * The callback fired when any children Checkbox is checked or unchecked
+   */
+  onChange?: (value: string[]) => void
+}
+
+export const CheckboxGroup = forwardRef<HTMLDivElement, CheckboxGroupProps>(
+  (
+    {
+      name: nameProp,
+      value: valueProp,
+      defaultValue,
+      className,
+      intent,
+      orientation = 'vertical',
+      onChange: onChangeProp,
+      children,
+      ...others
+    },
+    ref
+  ) => {
+    const [value, setValue] = useCombinedState(valueProp, defaultValue)
+    const field = useFormFieldState()
+    const onChangeRef = useRef(onChangeProp)
+
+    const { id, labelId, description, isInvalid, isRequired } = field
+    const name = nameProp ?? field.name
+
+    const current = useMemo(() => {
```

Comment:
Is the usage of `useMemo` relevant here, given the large number of dependencies? It's unclear what computation we're avoiding here 😅

Reply:
If I don't do it every time the `CheckboxGroup` is rendered the context value will be a different object and `memo` won't work in consumer components. Nevertheless, this could be premature optimisation 🤔 WDYT ?

Some readings about the topic:
- https://blog.agney.dev/useMemo-inside-context/
- https://kentcdodds.com/blog/how-to-optimize-your-context-value

Reply:
> If I don't do it every time the CheckboxGroup is rendered the context value will be a different object

Yes, but with the current implementation, since the `useMemo` is dependent on the `value` variable, the context object will still be different every time the CheckboxGroup is rendered I'm afraid 😅

I mean, every time a checkbox will be checked / unchecked:
1. this will update the `value`
2. the `useMemo` will be recomputed / hence a new object will be created
3. a re-render will be triggered

Reply:
Yes! I totally forgot that it was an array while reading the comment. You are right!

Reply:
Yes, I'm not so sure that memoizing this makes sense, but it's no blocker for me either 🤷